packages/core/content-releases/admin/src/pages/ReleasesPage.tsx

```diff
@@ -1,25 +1,104 @@
 import * as React from 'react';
 
-import { Button, HeaderLayout } from '@strapi/design-system';
-import { CheckPermissions } from '@strapi/helper-plugin';
+import {
+  Box,
+  Button,
+  ContentLayout,
+  Flex,
+  HeaderLayout,
+  Main,
+  Tab,
+  TabGroup,
+  TabPanel,
+  TabPanels,
+  Tabs,
+} from '@strapi/design-system';
+import {
+  CheckPermissions,
+  LoadingIndicatorPage,
+  PageSizeURLQuery,
+  PaginationURLQuery,
+  useQueryParams,
+} from '@strapi/helper-plugin';
 import { Plus } from '@strapi/icons';
 import { useIntl } from 'react-intl';
 
 import { AddReleaseDialog } from '../components/AddReleaseDialog';
+import { ReleasesGrid } from '../components/ReleasesGrid';
 import { PERMISSIONS } from '../constants';
+import { useGetReleasesQuery } from '../modules/releaseSlice';
+
+interface ReleasesPageQuery {
+  page?: number;
+  pageSize?: number;
+  filters?: {
+    $and?: Array<{
+      releasedAt?: {
+        $notNull?: boolean;
+      };
+    }>;
+  };
+}
 
 const ReleasesPage = () => {
   const [addReleaseDialogIsShown, setAddReleaseDialogIsShown] = React.useState(false);
+  const [tabSelected, setTabSelected] = React.useState<'pending' | 'done'>('pending');
   const { formatMessage } = useIntl();
+  const [{ query }, setQuery] = useQueryParams<ReleasesPageQuery>();
+  const response = useGetReleasesQuery(query);
 
-  const total = 0; // TODO: replace it with the total number of releases
+  const { isLoading, isSuccess, isFetching, isError } = response;
 
   const toggleAddReleaseDialog = () => {
     setAddReleaseDialogIsShown((prev) => !prev);
   };
 
+  let totalEntries = 0;
+
+  if (isSuccess && !isFetching && response.currentData?.pagination) {
+    totalEntries = response.currentData?.pagination?.total;
+  }
```

Comment:
✨ code style comment: I think this should be one block of code since it is all part of the same idea.

```suggestion
  let totalEntries = 0;
  if (isSuccess && !isFetching && response.currentData?.pagination) {
    totalEntries = response.currentData?.pagination?.total;
  }
```

it's a small difference but makes the code more readable in my opinion.

Reply:
Do we not handle a loading state? Would it not be possible to set this variable after we are sure we have the data? I would prefer to avoid a mutable variable.